bin/tx-prover/src/proxy/metrics.rs

```diff
@@ -0,0 +1,79 @@
+use std::sync::LazyLock;
+
+use prometheus::{
+    register_histogram, register_int_counter, register_int_counter_vec, register_int_gauge,
+    Histogram, IntCounter, IntCounterVec, IntGauge,
+};
+
+// QUEUE METRICS
+// ================================================================================================
+
+pub static QUEUE_SIZE: LazyLock<IntGauge> =
+    LazyLock::new(|| register_int_gauge!("queue_size", "Number of requests in the queue").unwrap());
+pub static QUEUE_LATENCY: LazyLock<Histogram> = LazyLock::new(|| {
+    register_histogram!(
+        "queue_latency",
+        "Time (in seconds) requests spend in the queue",
+        vec![0.1, 0.5, 1.0, 2.0, 5.0, 10.0]
+    )
+    .unwrap()
+});
+pub static QUEUE_DROP_COUNT: LazyLock<IntCounter> = LazyLock::new(|| {
+    register_int_counter!("queue_drop_count", "Number of requests dropped due to a full queue")
+        .unwrap()
+});
+
+// WORKER METRICS
+// ================================================================================================
+
+pub static WORKER_COUNT: LazyLock<IntGauge> =
+    LazyLock::new(|| register_int_gauge!("worker_count", "Number of workers").unwrap());
```

Comment:
nit (not 100% sure it applies so feel free to change it or ignore it altogether):
```suggestion
    LazyLock::new(|| register_int_gauge!("worker_count", "Number of  healthy and available workers").unwrap());
```

Reply:
I'm not quite convinced on the "available" part, because this metric includes busy workers. What should available mean here? I thinking in a different word.

Reply:
I think `Total number of workers` is the one that reflects the better what the metric means.